Python/chapter03/p01_three_in_one/miguelHx.py

```diff
@@ -17,137 +17,49 @@ class StackNode(Generic[T]):
-        self.current_node: Optional[StackNode[T]] = self.top
-        return
-    
-    def pop(self) -> T:
-        """
-        Removes the top item from the stack
-
-        Raises:
-            IndexError: raised when pop is attempted on empty stack
-
-        Returns:
-            int: The data at the top of the stack
-        """
-        if self.top is None:
-            raise IndexError('Stack is Empty.')
-        item = self.top.data
-        self.top = self.top.next
-        self.current_node = self.top
-        self.size -= 1
-        return item
-    
-    def push(self, item: int) -> None:
-        """
-        Adds an item to the top of the stack
-        Args:
-            item (int): data we want at the top of stack
-        """
-        t = StackNode(item, None)
-        t.next = self.top
-        self.top = t
-        self.current_node = self.top
-        self.size += 1
-
-    def peek(self) -> int:
-        """
-        Returns data at the top of the stack
-
-        Raises:
-            IndexError: [description]
-
-        Returns:
-            int: the value at the top of the stack
-        """
-        if self.top is None:
-            raise IndexError('Stack is Empty')
-        return self.top.data
-    
-    def __iter__(self) -> Iterator:
-        """
-        Builds a list of the current stack state.
-        For example, given the following stack:
-            3 -> 2 -> 1, where 3 is the top,
-        Expect:
-            [3, 2, 1]
-        Returns:
-            List[int]: list of integers
-        """
-        return self
-    
-    def __next__(self) -> T:
-        self.index += 1
-        if self.index == self.size or self.current_node is None:
-            self.index = -1
-            self.current_node = self.top
-            raise StopIteration
-        n: T = self.current_node.data
-        self.current_node = self.current_node.next
-        return n
-    
-    def __bool__(self) -> bool:
-        """
-        Returns True if and only if the
-        stack is empty, otherwise False.
-
-        Returns:
-            bool: True if empty, False otherwise
-        """
-        return self.size == 0
-    
-    def __str__(self):
-        if self.size == 0:
-            return '<Empty>'
-        values = []
-        n = self.top
-        while n.next is not None:
-            values.append(str(n.data))
-            n = n.next
-        values.append(str(n.data))
-        return '->'.join(values)
+@dataclass
+class StackInfo:
+    id: int
+    start: int
+    end: int
+    size: int
+    top_index: int
+    top_index_next: int
 
 
 class StackTrio:
 
     def __init__(self, stack_capacity = 7):
         self.num_stacks = 3
         self.stack_capacity = stack_capacity
+        first_stack_info = StackInfo(
+            1, # id
+            0, # start index
+            stack_capacity - 1, # end index
+            0, # size
+            0, # top index
+            -1 # top index next
```

Comment:
```suggestion
            id: 1,
            start_index: 0,
            end_index: stack_capacity - 1,
            size: 0,
            top_index: 0,
            top_next_index: -1
```

this way the comments are not needed anymore, _and_ the parameters can be reordered safely.

Reply:
This led to a syntax error.  Seems like Python 3 doesn't support that syntax when initializing an instance of a dataclass.

Maybe the best way for now is to use kwargs?  Updated to use kwargs here:  6f0999438665d195fe19ccc3ae4adb00f80a2388

Reply:
oh derp! i forgot it's with `=`, not `:`! this is a function call, not a dictionary u_u

(note to self: try out suggestions locally)

Reply:
No worries, is this PR ready to be merged?